Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -330,4 +330,163 @@ public TablesResource.InsertRequest makeInsertReq()
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        public string Table { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset to search. Can be a string, a Dataset, a DatasetReference, or a DatasetsData object.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ByValue)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to search. This field takes Dataset or DatasetRefrence objects.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        public DatasetReference Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be inserted. Also takes TableReferences.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Apis.Bigquery.v2.Data.Table.TableReference))]
+        public TableReference InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Forces deletion of non-empty tables and the data contained in them.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByValue:
+                    // No processing needed for ByValue parameter sets.
+                    break;
+                case ParameterSetNames.ByObject:
+                    Project = InputObject.ProjectId;
+                    DatasetId = InputObject.DatasetId;
+                    Table = InputObject.TableId;
+                    break;
+                case ParameterSetNames.ByDatasetObject:
+                    Project = Dataset.ProjectId;
+                    DatasetId = Dataset.DatasetId;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            TablesResource.DeleteRequest request = Service.Tables.Delete(Project, DatasetId, Table);
+            String response = "Table Removal Stopped.";
+            if (ShouldProcess(Table))
+            {
+                try
+                {
+                    if (Force == true)
```

Comment:
This is a personal preference, but I don't like `BoolVar == boolLiteral` when `BoolVar` or `!BoolVar` would do.